src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -278,4 +314,14 @@ private function addServiceToAmbiguousType($id, $type)
         }
         $this->ambiguousServiceTypes[$type][] = $id;
     }
+
+    static private function getResourceMetadataForMethod(\ReflectionMethod $method)
+    {
+        $methodArgumentsMetadata = array();
+        foreach ($method->getParameters() as $parameter) {
+            $methodArgumentsMetadata[] = (string) $parameter;
```

Comment:
And casting to a string probably give us a lot of noise like the default value, ... Why not just explicitly use what make sense for us here?